pkg/artifactory/resource/replication/replication.go

```diff
@@ -1,3 +1,55 @@
 package replication
 
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/client"
+)
+
 const EndpointPath = "artifactory/api/replications/"
+
+var replicationSchemaEnableEventReplication = map[string]*schema.Schema{
+	"enable_event_replication": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "When set, each event will trigger replication of the artifacts changed in this event. This can be any type of event on artifact, e.g. add, deleted or property change. Default value is `false`.",
+	},
+}
+
+func resourceReplicationDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	_, err := m.(*resty.Client).R().
+		AddRetryCondition(client.RetryOnMergeError).
+		Delete(EndpointPath + d.Id())
+	return diag.FromErr(err)
```

Comment:
Don't we need to call `d.SetId("")` to reset the resource ID? Like: https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/repository/repository.go#L175